test/widget/auth/email_auth_options_page_test.dart

```diff
@@ -329,50 +331,93 @@ void main() {
+      // Create Finders for text fields and create account button.
       final createAccountButton = find.byType(CreateAccountButton);
-      expect(createAccountButton, findsOneWidget);
       final emailTextField = find.byType(EmailTextField);
-      expect(emailTextField, findsOneWidget);
       final passwordTextField = find.byType(PasswordTextField);
-      expect(passwordTextField, findsOneWidget);
       final repeatPasswordTextField = find.byType(RepeatPasswordTextField);
+
+      // Check that all the expected widgets are present.
+      expect(createAccountButton, findsOneWidget);
+      expect(emailTextField, findsOneWidget);
+      expect(passwordTextField, findsOneWidget);
       expect(repeatPasswordTextField, findsOneWidget);
 
-      // Tap to validate form
+      // Create Finders for the text shown by the textfields when invalid.
+      final invalidEmailText = find.text('please enter a valid email');
+      final invalidPasswordText =
+          find.text('password must be between 6 and 30 characters');
+      final invalidRepeatPasswordText = find.text('passwords do not match');
+
+      // Input invalid details.
+      await tester.enterText(emailTextField, 'invalid_email');
+      await tester.enterText(passwordTextField, '123');
+      await tester.enterText(repeatPasswordTextField, '12345');
+
+      // Submit invalid form.
       await tester.tap(createAccountButton);
 
-      // Check that empty form is invalid
-      expect(testMiddleware.received(SignUpWithEmail()), false);
+      // Check that the widget dispatched the expected action.
+      expect(
+          fakeStore.dispatchedActions,
+          contains(UpdateEmailAuthOptionsPage(
+              autovalidate: AutoValidate.onUserInteraction)));
 
-      // Input valid email and password
+      // Update the state in response to the actions.
+      fakeStore.updateState((b) => b
+        ..emailAuthOptionsPage.autovalidate = AutoValidate.onUserInteraction);
+
+      await tester.pumpAndSettle();
+
+      // Check formfield error messages are shown.
+      expect(invalidEmailText, findsOneWidget);
+      expect(invalidPasswordText, findsOneWidget);
+      expect(invalidRepeatPasswordText, findsOneWidget);
+
+      // Input valid email and password.
       await tester.enterText(emailTextField, 'test@email.com');
       await tester.enterText(passwordTextField, 'password123');
       await tester.enterText(repeatPasswordTextField, 'password123');
 
-      // Tap to validate valid form
+      // Check that the widget dispatched the expected actions.
+      expect(
+          fakeStore.dispatchedActions,
+          containsAll(<ReduxAction>[
+            UpdateEmailAuthOptionsPage(email: 'test@email.com'),
+            UpdateEmailAuthOptionsPage(password: 'password123'),
+            UpdateEmailAuthOptionsPage(repeatPassword: 'password123')
+          ]));
+
+      // Update the state in response to the actions.
+      fakeStore.updateState((b) => b
```

Comment:
Updating the app state so the widgets rebuild with the new state.

Reply:
nice